services/graph.js

```diff
@@ -1,136 +1,230 @@
-      `${process.env.WRI_API_URL}/graph/query/list-concepts?application=${process.env.APPLICATIONS}&env=${process.env.API_ENV}`,
-      { headers: { 'Upgrade-Insecure-Requests': 1 } }
-
-    )
-      .then(response => response.json())
-      .then(response => response.data);
-  }
-  /**
-   * Get inferred tags
-   */
-  getInferredTags(tags) {
-    return fetch(
-      `${process.env.WRI_API_URL}/graph/query/concepts-inferred?concepts=${tags}&application=${process.env.APPLICATIONS}&env=${process.env.API_ENV}`,
-      { headers: { 'Upgrade-Insecure-Requests': 1 } }
-
-    )
-      .then(response => response.json())
-      .then(response => response.data);
+import WRISerializer from 'wri-json-api-serializer';
+
+// utils
+import { WRIAPI } from 'utils/axios';
+import { logger } from 'utils/logs';
+
+/**
+ * Get all tags
+ */
+export const fetchAllTags = (params = {}) => {
+  logger.info('Fetch all tags');
+  return WRIAPI.get('graph/query/list-concepts',
+    {
+      headers: { 'Upgrade-Insecure-Requests': 1 },
+      params: {
+        env: process.env.API_ENV,
+        application: process.env.APPLICATIONS,
+        ...params
+      }
+    })
+    .then(response => response.data.data)
+    .catch((response) => {
+      const { status, statusText } = response;
+      logger.error(`Error fetching all tags: ${status}: ${statusText}`);
+      throw new Error(`Error fetching all tags: ${status}: ${statusText}`);
+    });
+};
+
+/**
+ * Get inferred tags
+ */
+export const fetchInferredTags = (tags, params = {}) => {
+  logger.info('Fetch inferred tags');
+  return WRIAPI.get(`graph/query/concepts-inferred?concepts=${tags}`,
+    {
+      headers: { 'Upgrade-Insecure-Requests': 1 },
+      params: {
+        env: process.env.API_ENV,
+        application: process.env.APPLICATIONS,
+        ...params
+      }
+    })
+    .then(response => response.data.data)
+    .catch((response) => {
+      const { status, statusText } = response;
+      logger.error(`Error fetching inferred tags: ${tags} ${status}: ${statusText}`);
+      throw new Error(`Error inferred tags: ${tags} ${status}: ${statusText}`);
+    });
+};
+
+/**
+* Get dataset tags
+*/
+export const fetchDatasetTags = (datasetId, params = {}) => {
+  logger.info(`Fetch dataset tags: ${datasetId}`);
+  return WRIAPI.get(`dataset/${datasetId}/vocabulary`,
+    {
+      headers: { 'Upgrade-Insecure-Requests': 1 },
+      params: {
+        env: process.env.API_ENV,
+        application: process.env.APPLICATIONS,
+        ...params
+      }
+    })
+    .then(response => WRISerializer(response.data))
+    .catch((response) => {
+      const { status, statusText } = response;
+      logger.error(`Error fetching dataset tags ${datasetId}: ${status}: ${statusText}`);
+      throw new Error(`Error fetching dataset tags ${datasetId}: ${status}: ${statusText}`);
+    });
+};
+
+/**
+* Update dataset tags
+*/
+export const updateDatasetTags = (datasetId, tags, token, usePatch = false) => {
```

Comment:
I think this kind of function is pure evil. This one is able to trigger 3 different fetches based on some params. I won't enter too much in detail if we are going to rebuild the graph service, but this is the kind of flow we're running away.